api/service/prometheus/service.go

```diff
@@ -29,32 +51,83 @@ type Handler struct {
 }
 
 var (
-	svc = &Service{}
+	registryOnce sync.Once
+	svc          = &Service{}
+	config       Config
 )
 
+func getJobName(config Config) string {
+	var node string
+
+	// legacy nodes are harmony nodes: s0,s1,s2,s3
+	if config.legacy {
+		node = "s"
+	} else {
+		if config.nodetype == "validator" {
+			// regular validator nodes are: v0,v1,v2,v3
+			node = "v"
+		} else {
+			// explorer nodes are: e0,e1,e2,e3
+			node = "e"
+		}
+	}
+
+	return fmt.Sprintf("%s/%s%d", config.network, node, config.shard)
+}
+
 // NewService sets up a new instance for a given address host:port.
 // An empty host will match with any IP so an address like ":19000" is perfectly acceptable.
-func NewService(config nodeconfig.PrometheusServerConfig, additionalHandlers ...Handler) {
-	if !config.HTTPEnabled {
+func NewService(additionalHandlers ...Handler) {
+	if !config.enabled {
```

Comment:
Do you think enabled would better be decided in upper function calls?

Reply:
not really, better to keep all the config in one place though.